### ISSUE TITLE
make lib version of capirca

### ISSUE THREAD

**Issue by Ali-aqrabawi** [google/capirca#162]
any plans to have a library of capirca so it can be integrated with other existing applications ?

**Comment by ankenyr** [comment]
What do you mean? It is python code. All you need to do is put it in your path either manually or via a package manager and then import it. Do you mean bundling the lib modules without the scrips like aclgen or definate?

**Comment by Ali-aqrabawi** [comment]
basicly to have capirca to accept rules as json intead of `.pol` file

**Comment by ankenyr** [comment]
tl;dr: no, we will not be doing this.
If we were to switch over to using json then someone else may come along and want yaml, or xml, or some other format. aclgen.py and the parser are mostly just examples. If they work for you then great. If you want to create your own pipeline for automatically generating ACLs then you should implement your own that fits for you.

The generators are the bulk of this project and they all take a policy object which can be constructed via any method you want. If you want to create json files that represents those objects then that is a pretty quick script. If you have some crazy pipeline that pulls in dynamic information about your network at any given time and constructs the policy object on the fly, you can do that too.
